gnucash/gnome-utils/gnc-account-sel.c

```diff
@@ -144,33 +145,241 @@ combo_changed_cb (GNCAccountSel *gas, gpointer combo)
+{
+    GNCAccountSel *gas = GNC_ACCOUNT_SEL(user_data);
+    gas->hide_hidden = gtk_check_menu_item_get_active (GTK_CHECK_MENU_ITEM(widget));
+    update_entry_and_list (gas);
+}
+
+static void
+icon_release_cb (GtkEntry *entry, GtkEntryIconPosition icon_pos,
+                 GdkEvent* event, gpointer user_data)
+{
+    GNCAccountSel *gas = GNC_ACCOUNT_SEL(user_data);
+    GtkWidget *menu, *h_placeholder, *h_hidden;
+
+    if (icon_pos != GTK_ENTRY_ICON_SECONDARY)
+        return;
+
+    menu = gtk_menu_new ();
+    h_placeholder = gtk_check_menu_item_new_with_label (_("Hide Placeholder Accounts"));
+    gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM(h_placeholder), gas->hide_placeholder);
+    h_hidden = gtk_check_menu_item_new_with_label (_("Hide Hidden Accounts"));
+    gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM(h_hidden), gas->hide_hidden);
+    gtk_menu_shell_append (GTK_MENU_SHELL(menu), h_placeholder);
+    gtk_menu_shell_append (GTK_MENU_SHELL(menu), h_hidden);
+    gtk_widget_show_all (menu);
+
+    g_signal_connect (G_OBJECT(h_placeholder), "activate",
+                      G_CALLBACK(toggle_placeholder_cb), gas);
+    g_signal_connect (G_OBJECT(h_hidden), "activate",
+                      G_CALLBACK(toggle_hidden_cb), gas);
+
+    gtk_menu_popup_at_pointer (GTK_MENU(menu), (GdkEvent *)event);
+}
+
 static void
 gnc_account_sel_init (GNCAccountSel *gas)
 {
     GtkWidget *widget;
+    GtkWidget *entry;
+    GtkEntryCompletion *completion;
 
     gtk_orientable_set_orientation (GTK_ORIENTABLE(gas), GTK_ORIENTATION_HORIZONTAL);
 
     gas->initDone = FALSE;
     gas->acctTypeFilters = FALSE;
     gas->newAccountButton = NULL;
     gas->currentSelection = -1;
+    gas->hide_placeholder = TRUE;
+    gas->hide_hidden = TRUE;
 
     g_object_set (gas, "spacing", 2, (gchar*)NULL);
 
     // Set the name for this widget so it can be easily manipulated with css
     gtk_widget_set_name (GTK_WIDGET(gas), "gnc-id-account-select");
 
-    gas->store = gtk_list_store_new (NUM_ACCT_COLS, G_TYPE_STRING, G_TYPE_POINTER);
+    gas->store = gtk_list_store_new (NUM_ACCT_COLS, G_TYPE_STRING,
+                                     G_TYPE_POINTER, G_TYPE_STRING);
     widget = gtk_combo_box_new_with_model_and_entry (GTK_TREE_MODEL(gas->store));
     gas->combo = GTK_COMBO_BOX(widget);
     gtk_combo_box_set_entry_text_column (GTK_COMBO_BOX(widget), ACCT_COL_NAME);
     g_signal_connect_swapped (gas->combo, "changed",
                               G_CALLBACK(combo_changed_cb), gas);
     gtk_container_add (GTK_CONTAINER(gas), widget);
 
+    entry = gtk_bin_get_child (GTK_BIN(gas->combo));
+    gtk_entry_set_icon_from_icon_name (GTK_ENTRY(entry), GTK_ENTRY_ICON_SECONDARY,
+                                       "preferences-system-symbolic");
+    gtk_entry_set_icon_tooltip_text (GTK_ENTRY(entry), GTK_ENTRY_ICON_SECONDARY,
+                                     _("Set visibility of Placeholder and Hidden Accounts."));
```

Comment:
"Set the visibility of placeholder and hidden accounts."
1. Article was missing.
2. No capitalization in tooltips

Reply:
Changed.